dj/sql/parsing/ast.py

```diff
@@ -220,25 +230,61 @@ class Expression(Node):
     """an expression type simply for type checking"""
 
 
+@dataclass(eq=False)
+class Name(Node):
+    """
+    the string name specified in sql with quote style
+    """
+
+    name: str
+    quote_style: str = ""
+
+    def __str__(self) -> str:
+        return (
+            f"{self.quote_style}{self.name}{self.quote_style}"  # pylint: disable=C0301
+        )
+
+    def to_identifier(self) -> "Identifier":
+        """
+        transforms a single Name to a single item Identifier
+        """
+        return cast(Identifier, Identifier([self]).add_self_as_parent())
+
+    def __hash__(self) -> int:
+        return hash(self.name + self.quote_style)
+
+
+@dataclass(eq=False)
+class Identifier(Node):
+    """
+    Represents a sequence of names for something Named in a query
+    """
+
+    idents: List[Name]
+
+    def __str__(self) -> str:
+        return ".".join(str(ident) for ident in self.idents)
+
+    def __hash__(self) -> int:
+        return hash(Identifier)
+
+
 @dataclass(eq=False)  # type: ignore
 class Named(Expression):
     """An Expression that has a name"""
 
-    name: str
-    quote_style: str = ""
+    ident: Identifier
 
     @property
-    def quoted_name(self) -> str:
+    def name(self) -> str:
         """
-        get the name of the Named Node including the quotes if any
+        return the name of the named node
         """
-        return (
-            f"{self.quote_style}{self.name}{self.quote_style}"  # pylint: disable=C0301
-        )
+        return str(self.ident)
 
     def alias_or_name(self) -> str:
         """
-        get the name or alias of the node
+        get the alias of a node if it is the descendant of an alias otherwise get its own ident
```

Comment:
`s/ident/identifier` Only because my mind read that as identity at first...unless it actually is supposed to be shorthand for identity :D

Reply:
Oh I see, it did mean ident as in an instance of `Name`, right? Any reason why we don't just call that `names` everywhere, including for the attribute for the `Identifier` class? Does that clash with something else?

Reply:
I think the docstring should read something more clear even like "returns the string name of a parent alias if one exists, otherwise, return the node's own string name".

Reply:
And then about the name thing, something doesn't feel right but I can't explain it and I can't ignore the logic of your proposal so I think I'll make the change!

Reply:
I thought a little bit more to put my feelings to words and I'm thinking that when I see `names` in this scenario it feels like there's some uniqueness implied as though a list of names suggests that they're all things that refer to the node _independently_ as opposed to things that _altogether_ make up what you refer to the node as. Does that make sense?

Reply:
That's fair and I agree. I looked around to see if there's a naming pattern in the database space and found:

[Oracle](https://docs.oracle.com/cd/B19306_01/server.102/b14200/sql_elements008.htm) calls them identifier parts: "...If an identifier includes multiple parts separated by periods, then each attribute can be up to 30 bytes long..."

[IBM](https://www.ibm.com/docs/en/informix-servers/14.10?topic=expressions-using-dot-notation) seems to call each one an identifier but also calls them components: "Dot notation (sometimes called the membership operator) qualifies a SQL identifier with another SQL identifier of which it is a component. You separate the identifiers with the period (.) symbol."

[MySQL](https://dev.mysql.com/doc/refman/8.0/en/identifier-qualifiers.html) calls each item an identifier but describes the last identifier as the name and all of the items before it as qualifiers: "An unqualified name consists of a single identifier. A qualified name consists of multiple identifiers."

Let's keep it how you have it now and maybe see how we all refer to them in conversations. 😄